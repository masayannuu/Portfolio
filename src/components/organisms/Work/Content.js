import React from 'react'
import styled from 'styled-components'
import { Image } from 'reactbulma'

import ParagraphTitle from '../../atoms/Paragraph/ParagraphTitle'
import InfoTagGroup from '../../molecules/Content/InfoTagGroup'
import AppImage from '../../../../public/dataoctopus.png'

const WrapDiv = styled.div`
  width: 591px;
`

const StyledTitle = styled(ParagraphTitle)`
  margin-bottom: 0.1rem;
`

const StyledSubTitle = styled(ParagraphTitle)`
  margin-bottom: 1rem;
`

const Content = ({ heading, sub_heading, info_tags }) => (
  <WrapDiv className="columns">
    <div className="column">
      <StyledTitle className="is-size-5 has-text-right">{heading}</StyledTitle>
      <StyledSubTitle className="is-size-6 has-text-right">{sub_heading}</StyledSubTitle>
      {info_tags.map(info_tag => <InfoTagGroup name={info_tag.name} tags={info_tag.tags} />)}
    </div>
    <div className="column">
      <Image src={AppImage} />
    </div>
  </WrapDiv>
)

export default Content
