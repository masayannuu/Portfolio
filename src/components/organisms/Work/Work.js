import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Image } from 'reactbulma'

import { Headline } from '../../atoms/Title/index'
import WorkContent from '../../molecules/Content/WorkContent'
import StyledSection from '../../atoms/Section/StyledSection'
import WorkImage from '../../../../public/work-side.png'

const StyledImage = styled(Image)`
  @media screen and (min-width: 300px) {
    width: 100%;
  }
  @media screen and (min-width: 600px) {
    position: -webkit-sticky;
    position: sticky;
    top: 50px;
    width: 367px;
    height: 593px;
  }
`

const Work = ({ contents }) => (
  <StyledSection name="work">
    <Headline>| WORK</Headline>
    <div className="columns">
      <div className="column">
        <StyledImage src={WorkImage} />
      </div>
      <div className="column">
        {contents.map(content => (
          <WorkContent
            content={content}
            key={
              content.title +
              Math.random()
                .toString(36)
                .slice(-8)
            }
          />
        ))}
      </div>
    </div>
  </StyledSection>
)

Work.propTypes = {
  contents: PropTypes.array.isRequired
}

export default Work
