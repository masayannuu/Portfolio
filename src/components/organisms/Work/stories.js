import React from 'react'
import { storiesOf } from '@storybook/react'

import Work from './Work'

const tags = ['Ruby', 'Js', 'java', 'C#', 'hoge']
const info_tags = [{ name: 'Language', tags }, { name: 'Use Tech', tags }]
const description =
  'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字'
const team_data = {
  number: 3,
  roll: '設計・開発主担当',
  work:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ'
}
const details = {
  title: 'File Transfer Service',
  sub_title: 'ファイル転送サービス',
  image_url: 'https://bulma.io/images/placeholders/1280x960.png',
  description,
  team_data,
  info_tags
}

const tests = [details, details, details, details, details]

storiesOf('Organisms.Work', module).add('Work', () => <Work contents={tests} />)
