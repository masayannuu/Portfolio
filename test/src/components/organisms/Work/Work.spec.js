import React from 'react'
import Enzyme from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Work from '../../../../../src/components/Organisms/Work/Work'

describe('Work Work Molecules', () => {
  it('should render Work', () => {
    const wrapper = Enzyme.shallow(<Work />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
