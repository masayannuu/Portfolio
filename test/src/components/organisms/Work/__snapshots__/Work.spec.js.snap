// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Work Work Molecules should render Work 1`] = `
<StyledSection
  id="work"
>
  <Headline>
    | WORK
  </Headline>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <Work__StyledImage
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
    <div
      className="column"
    >
      <Work__StyledLoading>
        Now Loading...
      </Work__StyledLoading>
    </div>
  </div>
</StyledSection>
`;
